cnudie/util.py

```diff
@@ -292,3 +294,29 @@ def enumerate_group_pairs(
             _add_if_not_duplicate(resource_diff.resource_refs_only_right, i)
 
     return resource_diff
+
+
+CTF_CD_FILENAME = 'component-descriptor.yaml'
+
+
+def extract_component_descriptor_from_ctf_archive(
+    path_to_ctf_tarfile: str
+) -> cm.ComponentDescriptor:
+    '''Extract the component descriptor from a CTF-tarfile
+
+    Only supports simple CTF-tarfiles containing one component at the moment.
+    '''
+    with tarfile.open(path_to_ctf_tarfile) as ctf_tar:
+        members = ctf_tar.getmembers()
+        if len(members) > 1:
+            raise RuntimeError('Only CTF archives with one component are supported')
+        component_tar = members[0]
+
+        # manually check whether member is a file to be able to generate a more expressive
+        # error-msg
+        if not component_tar.isfile():
```

Comment:
did we define ctc-archive(-tarfiles) to only contain files? if so, this check is of course valid. if not, we might want to skip entries that are not files